dan/webserver/webserver.py

```diff
@@ -9,6 +9,8 @@
 # store messages and current clients
 messages = []
 clients = set()
```

Comment:
We can get rid of this set this since you're adding the websocket to client_usernames. I'd suggest refactoring the code to just use the new client_usernames dict instead.